src/redux/actions/progressions/synchronize.js

```diff
@@ -28,65 +30,62 @@ export type Action =
       meta: {|id: string|}
     |}>;
 
-export const synchronizeProgression = (progressionId: string): StoreAction<Action> => {
-  return async (dispatch, getState, options) => {
-    await dispatch({
-      type: SYNCHRONIZE_REQUEST,
-      meta: {id: progressionId}
-    });
+export const synchronizeProgressions: StoreAction<Action> = async (dispatch, getState, options) => {
+  const {services} = options;
+  const state = getState();
+  const token = getToken(state);
+  const brand = getBrand(state);
 
-    const state = getState();
-    const token = getToken(state);
-    const brand = getBrand(state);
+  if (token === null) throw new TypeError('Token not defined');
+  if (brand === null) throw new TypeError('Brand not defined');
 
-    const {services} = options;
+  const progressions = await services.Progressions.getAll();
+  const synchronizedProgressionsIds: Array<string> = await services.Progressions.getSynchronizedProgressionIds();
 
+  const syncProgression = async (progression: Progression, countNumberOfRetries?: number = 0) => {
     try {
-      if (token === null) throw new TypeError('Token not defined');
-      if (brand === null) throw new TypeError('Brand not defined');
+      await services.Progressions.synchronize(token, brand.host, progression);
+      // $FlowFixMe here the progression will always have an id cause we check before calling the function
+      synchronizedProgressionsIds.push(progression._id);
 
-      const progression = await services.Progressions.findById(progressionId);
-      if (progression) {
-        await services.Progressions.synchronize(token, brand.host, progression);
+      return Promise.resolve();
+    } catch (error) {
+      if (error instanceof ForbiddenError) throw new Error(error);
+      if (countNumberOfRetries < 5) {
+        return syncProgression(progression, countNumberOfRetries + 1);
       }
-
-      return dispatch({
-        type: SYNCHRONIZE_SUCCESS,
-        meta: {id: progressionId}
-      });
-    } catch (err) {
-      return dispatch({
-        type: SYNCHRONIZE_FAILURE,
-        error: true,
-        payload: err,
-        meta: {id: progressionId}
-      });
+      return Promise.reject(error);
```

Comment:
```suggestion
      return throw error;
```

SI on utilise async/await, on est dans un paradigme imperatif, il faut alors throw une erreur et non retourner une Promise.